proposals/0031-astro-watch-check.md

```diff
@@ -0,0 +1,96 @@
+<!--
+  Note: You are probably looking for `stage-1--discussion-template.md`!
+  This template is reserved for anyone championing an already-approved proposal.
+
+  Community members who would like to propose an idea or feature should begin
+  by creating a GitHub Discussion. See the repo README.md for more info.
+
+  To use this template: create a new, empty file in the repo under `proposals/${ID}.md`.
+  Replace `${ID}` with the official accepted proposal ID, found in the GitHub Issue
+  of the accepted proposal.
+-->
+
+- Start Date: 2023-02-27
+- Reference Issues: https://github.com/withastro/roadmap/issues/473
+- Implementation PR: https://github.com/withastro/astro/pull/6356
+
+# Summary
+
+Add a `--watch` flag to `astro check` command.
+
+# Background & Motivation
+
+From the relevant issue:
+
+> In development mode there isn't a way to know if your .astro files contain no errors,
+> aside from editor feedback.
+>
+> If you use astro check you will know about errors, but might not see them until CI unless you remember to run the command before pushing.
+
+# Goals
+
+- A way to check TS errors in .astro files during development workflow.
+
+# Non-Goals
+
+- Having errors reported in the UI. This could be a nice enhancement in the future but would come with some downsides (such as possibly slowing down dev) that we want to punt on.
+
+# Detailed Design
+
+The new feature will use the existing tools that `astro` has. `vite` exposes a [`watcher` option](https://vitejs.dev/config/server-options.html#server-watch)
```

Comment:
Should be updated with the new `chokidar` method and the rationales behind going that way instead of using Vite